components/script/dom/document.rs

```diff
@@ -3190,8 +3187,8 @@ impl DocumentMethods for Document {
 
         // Step 2.
         // TODO: handle throw-on-dynamic-markup-insertion counter.
-
-        if !self.is_active() {
+        // FIXME: this should check for being active rather than fully active
```

Comment:
The problem is that we're currently not sending enough information from the constellation to the script thread to determine if a document is active. We're sending freeze/thaw messages to determine if a document is fully active, but not any messages about being active. I think this is a spec issue, the spec should really say "fully active" here.